src/app/worker/suggestion_processor.py

```
@@ -0,0 +1,323 @@
+)
+
+logger = get_logger(__name__)
+
+
+class NonRetryableJobError(Exception):
+    """Raised for failures that retrying cannot fix (e.g. a 4xx from
+    fluent-api, or a permanently malformed request payload). Jobs that
+    raise this go straight to 'failed' without consuming retry attempts."""
+
+
+async def process_job(
```

Ruff (0.15.20)

[warning] Too many branches (14 > 12)

(PLR0912)